testbed/tests/log_test.go

```diff
@@ -233,3 +241,116 @@ func TestLogOtlpSendingQueue(t *testing.T) {
+		name         string
+		sender       testbed.DataSender
+		receiver     testbed.DataReceiver
+		loadOptions  testbed.LoadOptions
+		sleepSeconds int
+	}{
+		{
+			name:     "filelog-largefiles-2Gb-lifetime",
+			sender:   datasenders.NewFileLogWriter(),
+			receiver: testbed.NewOTLPDataReceiver(testutil.GetAvailablePort(t)),
+			loadOptions: testbed.LoadOptions{
+				DataItemsPerSecond: 200000,
+				ItemsPerBatch:      1,
+				Parallel:           100,
+			},
+			sleepSeconds: 100,
+		},
+		{
+			name:     "filelog-largefiles-6GB-lifetime",
+			sender:   datasenders.NewFileLogWriter(),
+			receiver: testbed.NewOTLPDataReceiver(testutil.GetAvailablePort(t)),
+			loadOptions: testbed.LoadOptions{
+				DataItemsPerSecond: 330000,
+				ItemsPerBatch:      10,
+				Parallel:           10,
+			},
+			sleepSeconds: 200,
+		},
+		{
+			name:     "filelog-largefiles-50MB/sec",
+			sender:   datasenders.NewFileLogWriter(),
+			receiver: testbed.NewOTLPDataReceiver(testutil.GetAvailablePort(t)),
+			loadOptions: testbed.LoadOptions{
+				DataItemsPerSecond: 400000,
+				ItemsPerBatch:      100,
+				Parallel:           1,
+			},
+			sleepSeconds: 100,
+		},
+	}
+	processors := map[string]string{
+		"batch": `
+  batch:
+`}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ScenarioLong(
+				t,
+				test.sender,
+				test.receiver,
+				test.loadOptions,
+				performanceResultsSummary,
+				test.sleepSeconds,
+				processors,
+			)
+		})
+	}
+}
+
+func TestLargeFileOnce(t *testing.T) {
+	processors := map[string]string{
+		"batch": `
+  batch:
+`,
+	}
+	resultDir, err := filepath.Abs(path.Join("results", t.Name()))
+	require.NoError(t, err)
+	sender := datasenders.NewFileLogWriter()
+	receiver := testbed.NewOTLPDataReceiver(testutil.GetAvailablePort(t))
+	loadOptions := testbed.LoadOptions{
+		DataItemsPerSecond: 1,
+		ItemsPerBatch:      10000000,
+		Parallel:           1,
+	}
+
+	// Write data at once, before starting up the collector
+	dataProvider := testbed.NewPerfTestDataProvider(loadOptions)
+	dataItemsGenerated := atomic.Uint64{}
+	dataProvider.SetLoadGeneratorCounters(&dataItemsGenerated)
+	ld, _ := dataProvider.GenerateLogs()
+
+	m := &plog.ProtoMarshaler{}
+	fmt.Println(m.LogsSize(ld))
+	require.NoError(t, sender.ConsumeLogs(context.Background(), ld))
+	agentProc := testbed.NewChildProcessCollector(testbed.WithEnvVar("GOMAXPROCS", "2"))
+
+	configStr := createConfigYaml(t, sender, receiver, resultDir, processors, nil)
+	configCleanup, err := agentProc.PrepareConfig(configStr)
+	require.NoError(t, err)
+	defer configCleanup()
+
+	tc := testbed.NewTestCase(
+		t,
+		dataProvider,
+		sender,
+		receiver,
+		agentProc,
+		&testbed.CorrectnessLogTestValidator{},
+		performanceResultsSummary,
+	)
+	defer tc.Stop()
+
+	tc.StartBackend()
+	tc.StartAgent()
+
+	tc.WaitForN(func() bool { return dataItemsGenerated.Load() == tc.MockBackend.DataItemsReceived() }, 20*time.Second, "all logs received")
+
+	tc.StopAgent()
```

Comment:
What is the difference between this and Stop() ?

Reply:
`Stop()` collectively stops all the components and records the test results.

This particular `tc.StopAgent()` seems redundant as `Stop()` would handle it, hence I've removed it.